src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertyAccessorSymbol.cs

```diff
@@ -210,21 +210,15 @@ internal override bool IsExpressionBodied
             bool modifierErrors;
             var declarationModifiers = this.MakeModifiers(syntax, location, diagnostics, out modifierErrors);
 
-            // Include modifiers from the containing property.
-            propertyModifiers &= ~DeclarationModifiers.AccessibilityMask;
+            // Include some modifiers from the containing property.
+            propertyModifiers &= ~(DeclarationModifiers.AccessibilityMask | DeclarationModifiers.ReadOnly);
             if ((declarationModifiers & DeclarationModifiers.Private) != 0)
             {
                 // Private accessors cannot be virtual.
                 propertyModifiers &= ~DeclarationModifiers.Virtual;
             }
             declarationModifiers |= propertyModifiers & ~DeclarationModifiers.Indexer;
 
-            // auto-implemented struct getters are implicitly 'readonly'
-            if (containingType.IsStructType() && !property.IsStatic && isAutoPropertyAccessor && methodKind == MethodKind.PropertyGet)
-            {
-                declarationModifiers |= DeclarationModifiers.ReadOnly;
-            }
```

Comment:
Why remove this and override `IsDeclaredReadOnly`? If we revert this, could we marked `SourceMemberMethodSymbol.IsDeclaredReadOnly` as `sealed`?